dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/JspOptions.java

```diff
@@ -1072,9 +1092,23 @@ public void setJdkSourceLevel(int jdkSourceLevel) {
         this.jdkSourceLevel = jdkSourceLevel;
     }
 
+    public void setJavaSourceLevel(int javaSourceLevel) {
+        int jmv = JavaInfo.majorVersion();
+        if (javaSourceLevel > jmv) {
+            // can not specify higher than running Java
+            logger.logp(Level.WARNING, CLASS_NAME, "setJavaSourceLevel", JspMessages.getMessage("jsp.javasourcelevel.warning", new Object[] { javaSourceLevel, jmv }));
```

Comment:
A warning makes sense our docs specify: `Alerts user to a condition that might cause problems in the future such as configuration issues, usage of default values, and environmental conditions` Since this is a configuration issue.